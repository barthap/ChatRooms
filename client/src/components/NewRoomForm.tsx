import React from 'react';

import 'bootstrap/dist/css/bootstrap.min.css';
import { NewRoom } from '../common/room';

interface Props {
  onSubmit?: (newRoom: NewRoom) => void;
  onCancel?: () => void;
  errorMessage?: string | null;
}

export default function NewRoomForm({ onSubmit, onCancel, errorMessage }: Props = {}) {
  const [name, setName] = React.useState('');
  const [description, setDescription] = React.useState('');

  const handleSubmit: React.FormEventHandler<HTMLFormElement> = event => {
    event.preventDefault();
    onSubmit && onSubmit({ name, description });
  };

  return (
    <form onSubmit={handleSubmit}>
      <h3>Add New Room</h3>

      <div className="mb-3">
        <label htmlFor="name" className="form-label">
          Name:
        </label>
        <input
          type="text"
          minLength={3}
          className="form-control"
          placeholder="Enter room name"
          id="name"
          name="name"
          value={name}
          onChange={e => setName(e.target.value)}
        />
      </div>
      <div className="mb-3">
        <label htmlFor="description" className="col-form-label">
          Description:
        </label>
        <input
          type="text"
          minLength={3}
          className="form-control"
          placeholder="Enter room description"
          id="description"
          name="description"
          value={description}
          onChange={e => setDescription(e.target.value)}
        />
      </div>
      {errorMessage && (
        <div className="mb-3">
          <p className="text-danger">{errorMessage && <small>{errorMessage}</small>}</p>
        </div>
      )}
      <div className="d-flex justify-content-end">
        <button className="btn btn-secondary me-3" onClick={onCancel}>
          Cancel
        </button>
        <button type="submit" className="btn btn-primary">
          Add Room
        </button>
      </div>
    </form>
  );
}
