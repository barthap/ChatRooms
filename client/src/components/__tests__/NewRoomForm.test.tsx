import { render, screen } from '@testing-library/react';
import React from 'react';

import NewRoomForm from '../NewRoomForm';

describe('<NewRoomForm />', () => {
  it('renders form snapshot without error message', () => {
    const loginForm = render(<NewRoomForm />);
    expect(loginForm).toMatchSnapshot();
  });
  it('displays error message', () => {
    const msg = 'Test error';
    render(<NewRoomForm errorMessage={msg} />);
    const msgElement = screen.getByText(msg);
    expect(msgElement).toBeInTheDocument();
  });
});
