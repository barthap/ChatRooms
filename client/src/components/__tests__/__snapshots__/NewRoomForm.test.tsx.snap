// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewRoomForm /> renders form snapshot without error message 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <h3>
          Add New Room
        </h3>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="name"
          >
            Name:
          </label>
          <input
            class="form-control"
            id="name"
            minlength="2"
            name="name"
            placeholder="Enter room name"
            type="text"
            value=""
          />
        </div>
        <div
          class="mb-3"
        >
          <label
            class="col-form-label"
            for="description"
          >
            Description:
          </label>
          <input
            class="form-control"
            id="description"
            name="description"
            placeholder="Enter room description"
            type="text"
            value=""
          />
        </div>
        <div
          class="d-flex justify-content-end"
        >
          <button
            class="btn btn-secondary me-3"
          >
            Cancel
          </button>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Create Room
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <h3>
        Add New Room
      </h3>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="name"
        >
          Name:
        </label>
        <input
          class="form-control"
          id="name"
          minlength="2"
          name="name"
          placeholder="Enter room name"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb-3"
      >
        <label
          class="col-form-label"
          for="description"
        >
          Description:
        </label>
        <input
          class="form-control"
          id="description"
          name="description"
          placeholder="Enter room description"
          type="text"
          value=""
        />
      </div>
      <div
        class="d-flex justify-content-end"
      >
        <button
          class="btn btn-secondary me-3"
        >
          Cancel
        </button>
        <button
          class="btn btn-primary"
          type="submit"
        >
          Create Room
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
